internal/logic_handlers/organizationlogic/service.go

```diff
@@ -0,0 +1,240 @@
+			StatusCode: http.StatusInternalServerError,
+			Err:        fmt.Errorf("commit: %w", err),
+		})
+	}
+
+	return organization, nil
+}
+
+func (i *Service) AddOrganization(ctx context.Context, params *model.CreateOrganization) (*model.Organization, error) {
+	if err := params.Validate(); err != nil {
+		return nil, errs.New(&errs.Cfg{
+			StatusCode: http.StatusBadRequest,
+			Err:        fmt.Errorf("validate: %w", err),
+		})
+	}
+
+	tx, err := i.cfg.TxProvider.Tx(ctx)
+	if err != nil {
+		return nil, errs.New(&errs.Cfg{
+			StatusCode: http.StatusInternalServerError,
+			Err:        fmt.Errorf("get db: %w", err),
+		})
+	}
+	defer tx.Rollback(ctx)
+
+	exists, err := i.cfg.Persistor.GetOrganizationByName(ctx, tx, params.Name)
+	if err != nil {
+		if !strings.Contains(err.Error(), sysconsts.ErrExpectedExactlyOneEntry) {
+			return nil, errs.New(&errs.Cfg{
+				StatusCode: http.StatusBadRequest,
+				Err:        fmt.Errorf("check organization unique: %w", err),
+			})
+		}
+	}
+	if exists != nil {
+		return nil, errs.New(&errs.Cfg{
+			StatusCode: http.StatusBadRequest,
+			Err:        fmt.Errorf("organization already exists"),
+		})
+	}
+
+	organization, err := i.cfg.Persistor.AddOrganization(ctx, tx, params)
+	if err != nil {
+		return nil, errs.New(&errs.Cfg{
+			StatusCode: http.StatusInternalServerError,
+			Err:        fmt.Errorf("create: %w", err),
+		})
+	}
+
+	if err = tx.Commit(ctx); err != nil {
+		return nil, errs.New(&errs.Cfg{
+			StatusCode: http.StatusInternalServerError,
+			Err:        fmt.Errorf("commit: %w", err),
+		})
+	}
+
+	return organization, nil
+}
+
+func (i *Service) ListOrganization(
+	ctx context.Context,
+	filter *model.OrganizationFilters,
+) (*model.PaginatedOrganizations, error) {
+	db, err := i.cfg.TxProvider.Db(ctx)
+	if err != nil {
+		return nil, errs.New(&errs.Cfg{
+			StatusCode: http.StatusInternalServerError,
+			Err:        fmt.Errorf("get db: %w", err),
+		})
+	}
+
+	paginated, err := i.cfg.Persistor.GetOrganizations(ctx, db, filter)
+	if err != nil {
+		return nil, errs.New(&errs.Cfg{
+			StatusCode: http.StatusInternalServerError,
+			Err:        fmt.Errorf("get organizations: %w", err),
+		})
+	}
+
+	return paginated, nil
+}
+
+func (i *Service) UpdateOrganization(ctx context.Context, params *model.UpdateOrganization) (*model.Organization, error) {
+	tx, err := i.cfg.TxProvider.Tx(ctx)
+	if err != nil {
+		return nil, errs.New(&errs.Cfg{
+			StatusCode: http.StatusInternalServerError,
+			Err:        fmt.Errorf("get db: %w", err),
+		})
+	}
+	defer tx.Rollback(ctx)
+
+	if err := params.Validate(); err != nil {
+		return nil, fmt.Errorf("validate: %w", err)
+	}
+
+	organization, err := i.cfg.Persistor.UpdateOrganization(ctx, tx, params)
+	if err != nil {
+		return nil, fmt.Errorf("update organization: %w", err)
+	}
+
+	return organization, nil
+}
+
+func (s *Service) DeleteOrganization(ctx context.Context, params *model.DeleteOrganization) error {
+	tx, err := s.cfg.TxProvider.Tx(ctx)
```

Comment:
no need to use a transaction here because delete will only do 1 operation , which will either succeed or fail.. So, just use db

Reply:
Understood. The function is no longer using a transactions and is now just using a db.